src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs

```diff
@@ -380,6 +380,42 @@ public virtual void EnableExtensions(bool enable = true)
             SqliteException.ThrowExceptionForRC(rc, _db);
         }
 
+        /// <summary>
+        ///     Backup of the connected database.
+        /// </summary>
+        /// <param name="destination">The destination of the backup.</param>
+        public virtual void BackupDatabase(SqliteConnection destination)
+            => BackupDatabase(destination, "main", "main");
+
+        /// <summary>
+        ///     Backup of the connected database.
+        /// </summary>
+        /// <param name="destination">The destination of the backup.</param>
+        /// <param name="destinationName">The name of the destination database.</param>
+        /// <param name="sourceName">The name of the source database.</param>
+        public virtual void BackupDatabase(SqliteConnection destination, string destinationName, string sourceName)
+        {
+            if (_db == null
+                || _db.ptr == IntPtr.Zero
+                || destination.Handle == null
+                || destination.Handle.ptr == IntPtr.Zero)
+            {
+                throw new InvalidOperationException(Resources.CallRequiresOpenConnection(nameof(BackupDatabase)));
+            }
+
+            var backup = raw.sqlite3_backup_init(destination.Handle, destinationName, _db, sourceName);
+            if (backup.ptr == IntPtr.Zero)
+            {
+                SqliteException.ThrowExceptionForDatabaseError(_db);
+            }
+
+            var rc = raw.sqlite3_backup_step(backup, -1);
+            SqliteException.ThrowExceptionForRC(rc, _db);
+
+            rc = raw.sqlite3_backup_finish(backup);
+            SqliteException.ThrowExceptionForRC(rc, _db);
```

Comment:
No need to call `sqlite3_backup_finish`, just wrap `backup` inside a `using` statement.

Reply:
Okay, changed.